_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,341 @@
+retrun (
+    ...
+   <EuiHorizontalRule />
+   {DataSourceSelectorComponent}
+   <EuiHorizontalRule />
+     <EuiTextColor>
+       Selected: {selectedDataSources.map((ds) => ds.label).join(', ')}
+   </EuiTextColor>
+   <EuiHorizontalRule />
+   <EuiButton type="primary" size="s" onClick={onClickHandler}>
+     <FormattedMessage id="mayTest.buttonText" defaultMessage="Get Indices" />
+   </EuiButton>
+   <p>
+     <FormattedMessage
+       id="blogTest.indicesText"
+       defaultMessage="Get indices: {data}"
+       values={{ data: indices ? indices : 'Unknown' }}
+     />
+   </p>
+  ...
+)
+```
+
+Congratulations! Now you can run your plugins and use the pickers to select the desired data source. 
+Get indexes from remote cluster call datasource1:
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/blog_example_remote_cluster.png">
+
+Get indexes from a local cluster:
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/blog_example_local_cluster.png">
+
+For additional details on the example plugin code, please refer to [this repo](https://github.com/zhyuanqi/OpenSearch-Dashboards/tree/blog_example/examples/blog_example_plugin). 
+
+## 2 Developing, testing, and debugging
+
+Now that you've learned how to integrate multiple data sources with Dashboards plugins, let's explore some advanced knowledge and skills.
+
+### 2.1 Example plugins for MDS
+
+To help developers get started quickly with the components, we've included an example plugin for a more intuitive and direct experience. You can run Dashboards with the CLI below to enable the example plugins: 
+```
+yarn start --run-examples
+```
+
+Once Dashboards is up and running, navigate to the page and click on the menu on the left. From there, open the **developer examples** under OpenSearch Dashboards. Then click on **Multiple Data Sources Integration**. On the right side of the page, you'll find a comprehensive list of pickers, each accompanied by a brief introduction and a set of properties, listed below. Feel free to experiment with the example plugin. 
+
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/multi_selectable.png">
+
+### 2.2 The data source as a saved object
+
+The data source is persisted as a saved object in the Dashboards system index. At the time of creation, a UUID is assigned as its unique identifier and data source metadata is also gathered, including the remote cluster version and associated plugins. This automatically generated data, as well as user-provided information including the title, endpoint, and authentication methods, are stored as attributes of the data source object. For the current schema of the object, refer to the [related code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/common/data_sources/types.ts#L8). 
+
+## 3 FAQ 
```

Comment:
```suggestion
## Frequently asked questions 
```

Reply:
I suggest moving the FAQs to after the plugin list section.

Reply:
Got it.